### DISCUSSION TITLE
actions-runner-controller is moving to a new home!

### DISCUSSION THREAD

**Discussion by Link-** [actions/actions-runner-controller#2072] [Announcements]
Hey folks - just wanted to let you all know that after chatting with @mumoshu, GitHub has agreed to adopt the Action Runner Controller project and we will be starting work to **turn this into a full GitHub product**.

Since the launch of Actions, there have been requests for GitHub to support auto scaling of self hosted runners. Over the last year we have been reviewing this and talking to customers about their needs, many expressed that their primary goal was Linux workloads and that having something that was platform independent was key. We also heard a lot of customers sharing that they were using Actions-Runner-Controller, so we started working with @mumoshu and the rest of the maintainers earlier in the year.

We chose Actions-Runner-Controller so we had a platform agnostic, scalable solution for Linux users (which makes up a significant percentage of Actions use today). We have been incredibly lucky to have amazing contributions from the community and we're also very grateful to @toast-gear who has helped us enable this transition and will continue to work with us as we move forward with the project.

Looking ahead, we will be working on first **introducing a new auto scaling mode** which will hopefully eliminate some of the over and under scaling issues in the current approaches. Along with this, we will be starting our journey to provide **further test coverage** and eventually provide **full support for the project** for our Enterprise customers. 

To find out more, check out the roadmap here: https://github.com/orgs/github/projects/4247/views/1

**During the week of December 12** we will move the repository into the [github.com/actions](https://github.com/actions) organization in preparation for releasing fully supported Helm charts and containers built and maintained by GitHub.

#### What will happen to the published releases and containers?

We want to make sure this transition is as smooth as possible for everyone. Once we transfer this repository (as is) to the github.com/actions organization we will recreate this repository in this organization and republish the existing Helm charts. However, we will not be publishing new releases to the recreated repository.

#### Will the self-hosted runner images on Dockerhub remain?

They will remain where they are for now but will be sunset that repository in the future. We will communicate the location of the new container images for actions-runner-controller and self-hosted runners once we're ready to release the new auto scaling mode.

### Acknowledgements

This project would not have come this far without the countless hours volunteered by @summerwind @mumoshu @toast-gear and the wider community, you're all amazing and inspiring ❤️ our hope is that we continue to evolve this project together moving forward.

Credit is also due to the GitHub team who has been working really hard on the new scaling mode that will be released soon, huge shoutout to @TingluoHuang @nikola-jokic @fhammerl @cory-miller @AvaStancu @benwells @rentziass @vanzeben @johnsudol @thboop @Vijay-train @isethi @chrispat @nebuk89 and many others

**Comment by ap0phi5** [comment]
This is excellent news and will definitely help with wider enterprise adoption by giving this project the recognition it deserves! 💯

**Comment by reiniertimmer** [comment]
Fantastic news! ARC has been immensely valuable in our enterprise, and it has been a great tool to work with. Congratulations and thanks to @mumoshu @summerwind @toast-gear  and all the other contributors!

**Comment by RobertKelly** [comment]
Great news! Big thanks to the maintainers here. 👏🏻 🎉

**Comment by Link-** [comment]
ARC repository transfer to https://github/actions will be executed at `11:00AM CET today`

**Reply by Link-** [reply]
ARC repository transfer has started now: 11:00AM CET

**Reply by Link-** [reply]
Transfer has been completed successfully.

More details & action items can be found here: https://github.com/actions/actions-runner-controller/issues/2089#issuecomment-1346354012

**Reply by Link-** [reply]
1. Helm charts have been transparently moved to https://github.com/actions-runner-controller/actions-runner-controller.github.io and republished to the same URL https://actions-runner-controller.github.io/actions-runner-controller/
1. Mirror repository has been removed and redirections have been re-enabled all links pointing to https://github.com/actions-runner-controller/actions-runner-controller will now redirect to this repository

**Comment by danmanners** [comment]
This is amazing! Thanks to everyone who's been working on this, and I'm excited to see how this continues forward!

**Comment by matheensyedaslam** [comment]
Awesome!!!

**Comment by lorencarvalho** [comment]
Fantastic news!!

**Comment by ColinHeathman** [comment]
Awesome stuff!

**Comment by harshaisgud** [comment]
Excellent News!

**Comment by AlonAvrahami** [comment]
@Link- Can you provide us some info about the new auto scaling mode? very excited about this!
Also, now that this project is part of GitHub, are we going to see a roadmap for this project?

**Reply by Link-** [reply]
@AlonAvrahami within this month we will be merging the auto scaling mode to this repo, it'll contain all the information about how it works and what to expect from it! We're also doing private beta trials with a few customers this quarter to iron out the residual issues before making this available to everyone.

We will try as much as possible as possible to maintain open communication around our plans forward, whether the new features will make it to the public roadmap or not is yet to be determined

**Reply by AlonAvrahami** [reply]
@Link- What do i need to do in order to participate in the private beta?
We have a very demanding use-case which utilize more than 1000 runners at the same time, and the current webhook autoscaler isn't working properly :/
I would be more than happy to test it and help anyway i can!

**Reply by Link-** [reply]
@AlonAvrahami I'm afraid we have selected the private beta participants. There are multiple criteria we based our selection on, demand and scale were just one of them. We have to limit the beta to a select few because we're offering direct support and as any organisation we have limited resources. I understand your anticipation and it won't be long before this is made available for you to use!

**Comment by harshaisgud** [comment]
@Link- Will github be releasing official runner images with all of the language/application binaries in the self hosted runners now that it is taking over?

**Reply by Link-** [reply]
We are still discussing the pros / cons of doing this and we'll make an announcement about it

**Reply by harshaisgud** [reply]
Thanks @Link- Looking forward to it

**Comment by ajschmidt8** [comment]
Any updates on when the new autoscaling mode will be available to the general public?

My team at NVIDIA has been running into a lot of issues when using webhook-driven scaling with multiple organizations.

**Reply by ajschmidt8** [reply]
cc: @Link- ^^

**Reply by chrispat** [reply]
Hi Aj,

We are working through a limited beta of the new features now and hope to enable a public open beta in the near future.  Unfortunately, I do not have a specific date right now.

**Reply by pavel-napolov** [reply]
@ajschmidt8 maybe your managers at NVIDIA want to sponsor the project to make your team work easier?
